src/Plugins/DBFTPlugin/Consensus/ConsensusService.cs

```diff
@@ -95,9 +97,10 @@ private void InitializeConsensus(byte viewNumber)
                 else
                 {
                     TimeSpan span = neoSystem.Settings.TimePerBlock;
-                    if (block_received_index + 1 == context.Block.Index)
+                    if (block_received_index + 1 == context.Block.Index && onPrepareBlockIndex + 1 == context.Block.Index)
                     {
-                        var diff = TimeProvider.Current.UtcNow - block_received_time;
+                        // Include the consensus time into the consensus intervals.
+                        var diff = TimeProvider.Current.UtcNow - onPrepareReceivedTime;
```

Comment:
I believe this methodology can trigger strange behaviors in a real scenario, after change views, delays and etc...